python/paddle/v2/framework/layers.py

```diff
@@ -202,6 +190,30 @@ def concat(input, axis, program=None, init_program=None):
     return out
 
 
+def sums(input, program=None, init_program=None):
+    helper = LayerHelper('sum', **locals())
+    if not isinstance(input, list) and not isinstance(input, tuple):
+        input = [input]
+    out = helper.create_tmp_variable(dtype=input[0].data_type)
```

Comment:
`LayerHelper` has a member function: `input_dtype`. It will check whether all the inputs are the same type and then return the type. I think using it here is a better choice.
```python
def input_dtype(self, input_param_name='input'):
        inputs = self.multiple_input(input_param_name)
        dtype = None
        for each in inputs:
            if dtype is None:
                dtype = each.data_type
            elif dtype != each.data_type:
                raise ValueError("Data Type mismatch")
        return dtype
```